gym_solo/core/test_obs_observations.py

```diff
@@ -66,5 +66,157 @@ def test_compute(self, mock_vel, mock_base):
+    mock_joint_info.side_effect = joint_info_side_effect
+    dummy_robot_id = 0
+
+    o = obs.MotorEncoder(dummy_robot_id, degrees= degrees)
+
+    if degrees:
+      position_max = 572.9578
+      position_min = -572.9578
+    else:
+      position_max = 10
+      position_min = -10
+
+    lower_bound = np.full(num_joints, position_min)
+    upper_bound = np.full(num_joints, position_max)   
+
+    np.testing.assert_allclose(o.observation_space.low, lower_bound)
+    np.testing.assert_allclose(o.observation_space.high, upper_bound)
+
+
+  @mock.patch('pybullet.getJointInfo', autospec=True)
+  @mock.patch('pybullet.getNumJoints', autospec=True)
+  def test_labels(self, mock_num_joints, mock_joint_info):
+    num_joints = 12
+    dummy_robot_id = 0
+
+    mock_num_joints.return_value = num_joints
+    mock_joint_info.side_effect = self.joint_info
+    
+    ground_truth = ['FL_HFE', 'FL_KFE', 'FL_ANKLE', 'FR_HFE',
+     'FR_KFE', 'FR_ANKLE', 'HL_HFE', 'HL_KFE', 'HL_ANKLE',
+     'HR_HFE', 'HR_KFE', 'HR_ANKLE']
+
+    o = obs.MotorEncoder(dummy_robot_id)
+    self.assertEqual(o.labels(), ground_truth)
+    
+  
+  @parameterized.expand([
+    ("default", False),
+    ("degrees", True)
+  ])
+  @mock.patch('pybullet.getJointState', autospec=True)
+  @mock.patch('pybullet.getNumJoints', autospec=True)
+  def test_compute(self, name, degrees, mock_num_joints, mock_joint_state):
+    num_joints = 12
+    dummy_robot_id = 0
+
+    mock_num_joints.return_value = num_joints
+
+    # This is real case extracted from pybullet
+    mock_joint_state.side_effect = [
+      (1.5301299626083, 7.554435979113249e-11, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (-3.0853209964046426, -1.4807635358886225e-11, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (0.0, 0.0, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (1.530127327627307, 7.721743189663525e-12, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (-3.085315909474513, 1.2383787804023483e-11, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (0.0, 0.0, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (-1.530132288799807, -1.9996261740838904e-11, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (3.0853224548246283, -2.1458507061027347e-11, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (0.0, 0.0, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (1.5301292310246128, -2.8650758804247376e-11, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (-3.0853176193095613, 1.4177090742904175e-11, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0), 
+      (0.0, 0.0, (0.0, 0.0, 0.0, 0.0, 0.0, 0.0), 0.0)]
+    
+    # Built from the mock_joint_state.side_effect value. Refer to getJointState return from
+    # https://docs.google.com/document/d/10sXEhzFRSnvFcl3XxNGhnD4N2SedqwdAvK3dsihxVUA/edit#heading=h.p3s2oveabizm
+    ground_truth = [1.5301299626083, -3.0853209964046426, 0.0,
+      1.530127327627307, -3.085315909474513, 0.0,
+      -1.530132288799807, 3.0853224548246283, 0.0,
+      1.5301292310246128, -3.0853176193095613, 0.0]
```

Comment:
Kind of mentioned before, since we are testing a trivial computation, there really isn't a need to recreate perfect pybullet responses (especially cause all of the pybullet has already been mocked out anyways)

Reply:
I thought it was much better to test it on a real case rather than create a dummy one. Do you want me to change it to a dummy case?

Reply:
When you're writing unit tests, it's usually better fake out all real dependencies (and data) as you're literally tryna test the functionality of each method. But you're right, in *integration* testing, you def want to be using as much real data as possible.

We haven't started writing those tests yet (there aren't *too* many moving parts), but that's a case that we'll need to use real data because we'll interact with PyBullet and invalid configurations would cause flakes. In this case, where the observation is literally just unpacking a tuple, all of that data can get confusing.

It's your call if you want to change it or not. Since it's a unit test (all of PyBullet has been mocked out anyways), this isn't "real data" (if pybullet changes their API upstream, our test won't update) but rather more complicated dummy data. At the end of the day, it's a unit test that works, soooo... if it's too much effort to change it, I totally feel that! :P

Reply:
For now, I'll leave this here